### ISSUE TITLE
Fails to pass backupCapability check if mongodb is on remote server

### ISSUE THREAD

**Issue by nzalev** [Ylianst/MeshCentral#2486]
When hosting MongoDB on a remote server, meshcentral will fail to pass the `checkBackupCapability` test in `db.js`.

The following is the offending section (starting at line#1593 in db.js)
```javascript
            var mongoDumpPath = 'mongodump';
            if (parent.config.settings.autobackup && parent.config.settings.autobackup.mongodumppath) { mongoDumpPath = parent.config.settings.autobackup.mongodumppath; }
            const child_process = require('child_process');
            child_process.exec('"' + mongoDumpPath + '"', { cwd: backupPath }, function (error, stdout, stderr) {
                try {
                    if ((error != null) && (error != '')) {
                        if (parent.platform == 'win32') {
                            func(1, "Unable to find mongodump.exe, MongoDB database auto-backup will not be performed.");
                        } else {
                            func(1, "Unable to find mongodump, MongoDB database auto-backup will not be performed.");
                        }
```
By default, mongodump will eventually time out. If one concatenates error to the end of the error message, the following would be printed: 
```
WARNING: Unable to find mongodump, MongoDB database auto-backup will not be performed.Error: Command failed: "mongodump"
2021-04-07T17:22:07.148+0000    Failed: can't create session: could not connect to server: server selection error: server selection timeout, current topology: { Type: Single, Servers: [{ Addr:
 localhost:27017, Type: Unknown, State: Connected, Average RTT: 0, Last error: connection() : dial tcp [::1]:27017: connect: connection refused }, ] }
```

One *can* include the --uri option in the `mongoDumpPath` string under the `autoBackup` settings. However, doing so would conflict with the command string when actually performing the backup under `performBackup`, as it already includes the --uri option.
```javascript
                var mongoDumpPath = 'mongodump';
                if (parent.config.settings.autobackup && parent.config.settings.autobackup.mongodumppath) { mongoDumpPath = parent.config.settings.autobackup.mongodumppath; }
                const child_process = require('child_process');
                var cmd = '\"' + mongoDumpPath + '\" --db=\"' + dbname + '\" --archive=\"' + newBackupPath + '.archive\"';
                if (dburl) { cmd = '\"' + mongoDumpPath + '\" --uri=\"' + dburl.replace('?', '/?') + '\" --archive=\"' + newBackupPath + '.archive\"'; }
```

<br><br>

PS. The solution to this and #2423 have been fairly clear to me, but I am not sure if you accept pull requests from users (I know many maintainers do not). If you want, I can submit a fix.

**Comment by Ylianst** [comment]
Of course I accept pull requests! Thank you so much for looking into this and finding a solution. If you make a pull request, I will take it.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.8.5 that should include your fix.

**Comment by VINISHVV** [comment]
@nzalev & @Ylianst  - Thanks to fix this issue.
